### PR TITLE
More energy guns

### PR DESCRIPTION
**Pull request by Tryded**
## About the PR

Adds the IK-60, Miniature Energy Pistol, and PDW-9 Energy Pistol. IK-60 has 20 rounds now.
Miniature Energy Pistol can be ordered from cargo, the rest are researchable.
Added Research for all new energy guns
## Why / Balance

Colin said the guns were cool
## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/DeltaV-Station/Delta-v/assets/139474617/bac83a10-7253-4c55-ae89-4246c4a52b06

Video is outdated
Ignore the ERT members in the back
## Breaking changes

**Changelog**

:cl:Tryded
- add: Nanotrasen has recently introduced their miniature energy pistols and energy guns to the market and developed experimental research into the IK-60 laser carbine and the PDW-9 Energy Pistol. They can be researched by science.

### CODE REVIEW

**Comment by Tryded** [comment]
should be done now

**Comment by Tryded** [comment]
I have no idea why are these tests failing

**Comment by Colin-Tel** [comment]
YAML Linter is erroring on `/Locale/en-US/deltav/prototypes/catalog/cargo/cargo-armory.ftl`

**Comment by Tryded** [comment]
> YAML Linter is erroring on `/Locale/en-US/deltav/prototypes/catalog/cargo/cargo-armory.ftl`

I can't seem to figure out why it is failing, nor do I really know how ftls work

File: Resources/Locale/en-US/deltav/prototypes/catalog/cargo/cargo-armory.ftl [hunk 1]
```diff
@@ -6,3 +6,9 @@ ent-ArmoryUniversal = { ent-CrateArmoryUniversal }
 
 ent-ArmoryAdjutant = { ent-CrateArmoryAdjutant }
     .desc = { ent-CrateArmoryAdjutant.desc }
+
```
Comment (Colin-Tel) [inline Resources/Locale/en-US/deltav/prototypes/catalog/cargo/cargo-armory.ftl]:
```suggestion

```
There's an indent here that's probably tripping it.

**Comment by Tryded** [comment]
Take 2, found the extra indentation

**Comment by Tryded** [comment]
Okay, I actually don't know what is wrong now

**Comment by Colin-Tel** [comment]
> Okay, I actually don't know what is wrong now

image

**Comment by Tryded** [comment]
I just tested it in game and it works fine

**Comment by Tryded** [comment]
> > Okay, I actually don't know what is wrong now
> 
> image

I did this, it doesn't seem to be fixed

**Comment by Tryded** [comment]
Nevermind, it seems that I forgot to push the changes

**Comment by Tryded** [comment]
This should be it tbh

**Review by Colin-Tel — COMMENTED** [review]
Mini energy gun is size 10 and has 10 shots, also can be ordered at cargo.
PDW-9 is size 10 and has 20 shots
IK-60 is size 50 and has 20 shots
Energy Gun is size 50 and has 20 shots

all these weapons do the same amount of damage, what's the benefit of the bigger ones?

also it's worth mentioning that the mini and pdw are both size 10 even though the sprite for the pdw is like twice the size of the mini

**Comment by Tryded** [comment]
> Mini energy gun is size 10 and has 10 shots, also can be ordered at cargo. PDW-9 is size 10 and has 20 shots IK-60 is size 
50 and has 20 shots Energy Gun is size 50 and has 20 shots

The PDW-9 is requires a tier 2 research before it can be crafted, and it's recipe is quite expensive. It is meant to be a more efficient energy pistol, however I can balance it by lowering it's amount of shots a little bit and making normal energy gun higher.
Also, Energy guns can shoot 10 lethals **OR** 20 disablers. Keyword or. Same with the PDW-9
> all these weapons do the same amount of damage, what's the benefit of the bigger ones?

The IK-60 has a higher rate of fire and has more ammunition, however is locked behind an expensive tier 2 research and requires a lot of resources to craft. The PDW is also locked behind tier 2 research, it is basically a more efficient miniature energy pistol however is quite expensive to craft and research.

> also it's worth mentioning that the mini and pdw are both size 10 even though the sprite for the pdw is like twice the size of the mini

The sprites are just like that sadly, just like how the Mk58 is the size of a lecter, and I'm unsure how to fix that.

**Comment by Tryded** [comment]
> Mini energy gun is size 10 and has 10 shots, also can be ordered at cargo. PDW-9 is size 10 and has 20 shots IK-60 is size 50 and has 20 shots Energy Gun is size 50 and has 20 shots
> 
> all these weapons do the same amount of damage, what's the benefit of the bigger ones?
> 
> also it's worth mentioning that the mini and pdw are both size 10 even though the sprite for the pdw is like twice the size of the mini

I believe you misunderstand how energy guns work. 

in an energy gun, there is a total amount of power inside of the gun. Each shot depending on type uses different amount of charge.

For example, Miniature Energy gun has a total power of 1000. One Disabler shot uses 100 while Lethal uses 200. That means, it could fire 5 lethals or 10 disablers.

The PDW has a total power of 1600 after this commit, and each disabler cost 100 and lethal 200. It can fire 5 lethals OR 10 disablers before running out of energy.

The IK-60 Laser Carbine has a firerate of 180 rounds per minute (3 rounds per second) and has a power level of 4000. It can fire 20 laser rounds before exhausting its power supply

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml [hunk 2]
```diff
@@ -113,3 +113,147 @@
...
+    proto: BulletDisabler
+    fireCost: 100
+  - type: EnergyGun
+    fireModes:
+    - proto: BulletDisabler
+      fireCost: 200
+      name: disable
+      state: disabler
+    - proto: BulletEnergyGunLaser
+      fireCost:
```
Comment (Colin-Tel) [inline Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml]: isn't this missing a value?

**Comment by Colin-Tel** [comment]
> The IK-60 Laser Carbine has a firerate of 180 rounds per minute (3 rounds per second) and has a power level of 4000. It can fire 20 laser rounds before exhausting its power supply

This was the part I misunderstood. I forgot the 20 rounds were all lethal

File: Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml (hunk 2 above)
Comment (Tryded) [reply Resources/Prototypes/DeltaV/Entities/Objects/Weapons/Guns/Battery/battery_guns.yml]: Yeah it is missing, didn't see that

**Comment by Tryded** [comment]
Missing value should be fixed

**Comment by Tryded** [comment]
Any changes about balance that needs to be made?

**Comment by Colin-Tel** [comment]
> Any changes about balance that needs to be made?

Not from me. Debug said he wanted to review this prior to merge.

**Review by Colin-Tel — APPROVED** [review]
Debug left this at my discretion, so I'm going to approve it. We'll see how the players handle the fact these are researchable items.

**Pull request merged**